### ISSUE TITLE
Assert causes simulation to crash

### ISSUE THREAD

**Issue by niklwors** [OpenModelica/OpenModelica#11905]
### Description

This isue is related to #8200. But on windows this causes  the simulation to crash

### Steps to Reproduce
Simulate the model:

```modelica
model AssertModel
Real x(start=2);
equation
der(x) = 3*x;
assert(x<=10,"x becomes greater 10");
end AssertModel;
```
The simulation crashes with the error:
```
Simulation process failed. Exited with code 0xffffffffffffffff.
No event found, but assert was triggered. Throwing now!
```
and the simulation progres bar hangs.

### Version and OS

 - OpenModelica  v1.23.0-dev-214-gb890045362
 - OS:  Windows 10, 64 bit

**Comment by casella** [comment]
@niklwors the following notification is issued:
```
The following assertion has been violated at time 0.538000 ((x <= 10.0)) --> "x becomes greater 10"
```
then the simulation executable is not really crashing, it returns with error code -1 (which is reported as unsigned it for some reason). The progress bar stops at 54% of the StopTime-StartTime interval. This is different from simulations that end on purpose, like
```modelica
model TerminateModel
Real x(start=2);
equation
der(x) = 3*x;
when x> 0 then
  terminate("x has reached the value of 10");
end when;
end TerminateModel;
```
which terminate without an error code.

So, this is the expected behaviour, are you just annoyed by the red error message?

**Comment by niklwors** [comment]
Yes, it is the same problem. I had forgotten that I had reported it before. It's just happened again now.
